packages/gatsby-cli/src/handlers/plugin-add-utils.ts

```diff
@@ -22,19 +23,41 @@ const addPluginToConfig = (
     key: string
   }
 ): string => {
-  const addPlugins = new BabelPluginAddPluginsToGatsbyConfig({
-    pluginOrThemeName: name,
-    options,
-    shouldAdd: true,
-    key,
-  })
+  let code
 
-  // @ts-ignore - fix me
-  const { code } = transform(src, {
-    // @ts-ignore - fix me
-    plugins: [addPlugins.plugin],
-    configFile: false,
-  })
+  try {
+    const transformOptions: TransformOptions = {
+      plugins: [
+        [
+          BabelPluginAddPluginsToGatsbyConfig,
+          {
+            pluginOrThemeName: name,
+            options,
+            key,
+          },
+        ],
+      ],
+      filename: srcPath,
+      configFile: false,
+    }
+
+    // Use the Babel TS preset if we're operating on `gatsby-config.ts`
+    if (srcPath.endsWith(`ts`)) {
+      transformOptions.presets = [`@babel/preset-typescript`]
+    }
+
+    code = transform(src, transformOptions)?.code
+
+    // Add back stripped type import, do light formatting, remove added empty module export.
+    // Use semicolon since Babel does that anyway, and we might as well be consistent.
+    if (srcPath.endsWith(`ts`)) {
+      code = `import type { GatsbyConfig } from 'gatsby;'\n\n${code}`
```

Comment:
```suggestion
      code = `import type { GatsbyConfig } from "gatsby";\n\n${code}`
```

Snapshots also need an update then. (Semicolon was incorrectly placed)

If you look at the snapshots the single quote, double quote, backticks is a bit all over the place. I've switched this to double quotes now because that's also what we use in our docs.

Then the strings inside the config itself should be backticks. But doesn't have to be fixed in this PR

Reply:
Great catch, thanks!